### DISCUSSION TITLE
how to populate a styled dropdown via ajax

### DISCUSSION THREAD

**Discussion by tacman** [bigskysoftware/htmx#2899] [Q&A]
I'm searching a large list, and would like to populate a dropdown with the partial search results.  I guess something like https://github.com/afcapel/stimulus-autocomplete, it seems so natural that htmx would support it.

The closest I've found is https://htmx.org/examples/active-search/, but was wondering if there's an example for a dropdown.  Ideally a fancy one with multiple rows, images, styles, etc.  

Thanks.

**Comment by alainivars** [comment]
Have a look [here](https://turriate.com/articles/using-datalist-for-typeahead-htmx) for the origin
And [here](https://github.com/highfeature/highfeature_dashboard) for an example in a Django HTMLX POC application.
Sadly at the time, it is only populated by text, I did not managed to populate it with images and fancy styles, else we can have images with a [table](https://htmx.org/examples/active-search/) in place to a select or a datalist, but that does not update automatically the input text.
I will let you know if I manage to load image in it.

**Comment by dharmab** [comment]
I did it with:

- A server side endpoint that returns a partial results list through HTMX
- A bit of javascript to make the keyboard functionality a little nicer than the vanilla browser experience

https://github.com/dharmab/hyperboard/pull/56
